scout/adapter/mongo/variant.py

```diff
@@ -524,17 +545,31 @@ def check_causatives(self, case_obj=None, institute_obj=None, limit_genes=None):
             if other_case is None:
                 # Other variant belongs to a case that doesn't exist any more
                 continue
+
             other_link = var_event["link"]
             # link contains other variant ID
-            other_causative_id = other_link.split("/")[-1]
+            other_causative_id = other_link.split("/")[-1]  # md5-key ID of a variant
 
-            if other_causative_id in other_case.get("causatives", []):
-                positional_variant_ids.add(var_event["variant_id"])
+            if (
+                other_causative_id not in other_case.get("causatives", [])
+                and other_causative_id not in other_case.get("partial_causatives", {}).keys()
+            ):
+                continue
+
+            # Collect other causative variant's coords and change (exclude if it's clinical or research)
+            other_var_displ_name = var_event.get("subject").split("_")[
+                0:4
+            ]  # example: [ "17", "7577559", "G" "A"]
 
-            if other_causative_id in other_case.get("partial_causatives", {}).keys():
-                positional_variant_ids.add(var_event["variant_id"])
+            other_display_names.add("_".join(other_var_displ_name + ["clinical"]))
+            other_display_names.add("_".join(other_var_displ_name + ["research"]))
```

Comment:
Wouldn't it be faster and easier to md5sum these into variant_ids and uses as positional_variant_ids?

(I'm still annoyed by the docs btw; we have sooo many places where it says variant_id is just positional id, nothing about clinical/research type. Still, this version, with type in addition to the simple id is correct according to tests.)

Reply:
hehe that's how I did the PR in the beginning, but for some technical problem I've changed it. I'll check again!

Reply:
It wasn't that hard after all. I've changed the code to use md5 keys instead. Check it out when you have time!